src/pipelines/encore/encore-pipeline.ts

```diff
@@ -0,0 +1,163 @@
+    this.configuration = configuration;
+  }
+
+  async transcode(input: string, targetResolution: Resolution, targetBitrate: number, output: string, variables?: Record<string, string>): Promise<string> {
+    const response: Response = await this.createEncoreInstance()
+    logger.info(`createEncoreInstanceStatus: ${response.status}`);
+    logger.debug(`createEncoreInstanceBody: ${response.body}`);
+    const encoreInstance: EncoreInstance = JSON.parse(response.body)
+
+    while (true) {
+      await delay(5000);
+      const getResponse: Response = await this.getEncoreInstance(encoreInstance.url)
+      if (getResponse.status == 200) {
+        const jobResponse: Response = await this.createEncoreJob(input, encoreInstance.resources.enqueueJob.url)
+        logger.info(`createEncoreJobStatus: ${jobResponse.status}`);
+        logger.debug(`createEncoreJobBody: ${jobResponse.body}`);
+        break
+      }
+    }
+    // TODO: Poll or check jobs are done and have been analyzed. Then deleteEncoreInstance(instanceId: string).
+    return "todo"
+  }
+
+  async analyzeQuality(reference: string, distorted: string, output: string, model: QualityAnalysisModel): Promise<string> {
+    return "todo"
+  }
+
+  async createEncoreInstance(): Promise<Response> {
+    const url = 'https://api-encore.stage.osaas.io/encoreinstance';
+    const headers = [
+      'accept: application/json',
+      `x-jwt: Bearer ${this.configuration.token}`,
+      'Content-Type: application/json',
+    ];
+    const data = JSON.stringify({
+      "name": this.configuration.instanceId,
+      "profile": this.configuration.profile,
+    });
+
+    return new Promise<Response>((resolve) => {
+      const curl = new Curl();
+      curl.setOpt('URL', url);
+      curl.setOpt(Curl.option.HTTPHEADER, headers);
+      curl.setOpt(Curl.option.POSTFIELDS, data);
+
+      curl.on('end', function (statusCode: number, body: string | Buffer) {
+        resolve({ status: statusCode, body: body.toString() });
+        this.close();
+      });
+      curl.on('error', curl.close.bind(curl));
+      curl.perform();
+    })
+  }
+
+  async getEncoreInstance(instanceUrl: string): Promise<Response> {
+    const curl = new Curl();
+    const url = instanceUrl;
+    const headers = [
+      'accept: application/json',
+      `x-jwt: Bearer ${this.configuration.token}`,
+      'Content-Type: application/json',
+    ];
+
+    return new Promise<Response>((resolve) => {
+      const curl = new Curl();
+      curl.setOpt('URL', url);
+      curl.setOpt(Curl.option.HTTPHEADER, headers);
+      curl.setOpt(Curl.option.CUSTOMREQUEST, 'GET');
+
+      curl.on('end', function (statusCode: number, body: string | Buffer) {
+        resolve({ status: statusCode, body: body.toString() });
+        this.close();
+      });
+      curl.on('error', curl.close.bind(curl));
+      curl.perform();
+    })
+  }
+
+  async deleteEncoreInstance(instanceId: string): Promise<boolean> {
+    const curl = new Curl();
```

Comment:
Any particular reason to use curl and not `fetch` that are built-in in nodejs nowadays?

Reply:
No particular reason

Reply:
I would prefer if you can use `fetch` over curl. This to minimise bloat and 3rd party dependencies that we need to keep up to date.